Issue/1

0hun opened the pull request.

0hun committed:
feat (feature) : H2 설정

 H2 설정
 - H2 사용 설정
 - sql 보기 옵션 설정
 - h2 문법을 mysql로 변경

0hun committed:
feat (feature) : 회원 추가

 회원 추가
 - 회원 추가 기능 추가
 - 회원 관련 테스트 추가

0hun committed:
feat (feature) : 회원 추가

 회원 추가
 - 회원 추가 기능 추가
 - 회원 관련 테스트 추가

0hun committed:
Merge branch 'issue/1' of https://github.com/0hun/delivery-service into issue/1

yhmane commented on src/main/java/com/delivery/user/dto/UserDto.java:
```diff
@@ -0,0 +1,64 @@
+  // 상태 DEFAULT(기본), DELETED(삭제됨)
+  private DataStatus status;
+
+  public UserDto() {
```
lombok을 사용했으니 @NoArgsConstructor로 대신해도 될거 같아요!

0hun replied:
생성자 같은 경우 어노테이션을 사용안하고 구현을 했는데요. 롬복이 컴파일 타임에 바이트 코드를 끼워넣다보니 컴파일이 오래 걸릴 수 있어서, 생성자 만큼은 IDE를 통해서 생성하려고 합니다.

yhmane commented on src/main/java/com/delivery/user/service/UserService.java:
```diff
@@ -0,0 +1,26 @@
+
+  private final UserRepository userRepository;
+
+  public Optional<User> findById(long id) {
```
http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/

optinal에 대해서 잘 정리된 글이 있네용ㅎㅎ

yhmane commented on src/test/java/com/delivery/user/controller/UserControllerTest.java:
```diff
@@ -0,0 +1,78 @@
+    // then
+    final MvcResult mvcResult = resultActions.andExpect(status().isCreated()).andReturn();
+
+    assertThat(mvcResult.getResponse().getStatus()).isEqualTo(201);
```
201 check가 중복으로 이뤄지는거 같아요.

MockMvcRequestBuilders.post("/users/signUp")            .contentType(MediaType.APPLICATION_JSON)            .content(new Gson().toJson(userDto)))
.andExpect
로 상태값을 체크해도 좋을거 같습니다   😊

yhmane commented on src/test/java/com/delivery/user/domain/UserTest.java:
```diff
@@ -0,0 +1,30 @@
+    User user = userDto.toEntity();
+
+    //then
+    assertThat(user.getEmail()).isEqualTo(userDto.getEmail());
```
 👍

yhmane commented on src/test/java/com/delivery/user/dto/UserDtoTest.java:
```diff
@@ -0,0 +1,122 @@
+    //then
+    Set<ConstraintViolation<UserDto>> violations = validator.validate(userDto);
+
+    assertThat(violations).isEmpty();
```
👍👍

yhmane reviewed (commented):
리뷰용~

0hun committed:
refactor : 회원 추가 url 수정

 회원 추가 url 수정
 - 회원이라는 리소스를 추가하는 것이므로 rest api 표준에 맞게 url 수정
 - log4j 대신 logback을 사용하기 위해 인터페이스인 Slf4j로 로그 수정

0hun committed:
refactor : UserServiceTest 수정

 UserServiceTest 수정
 - user 저장 후 mock 객체를 리턴하도록 수정
 - 객체의 값을 꺼내서 비교하기 보다는 객체간 비교를 하도록 수정

0hun committed:
feat(test) : userEntityMapping test 추가

 userEntityMapping test 추가
 - user 객체의 값을 테스트하도록 추가
